include/qarchivememoryextractoroutput.hpp

```diff
@@ -1,24 +1,23 @@
 #ifndef QARCHIVE_MEMORY_EXTRACTOR_OUTPUT_HPP_INCLUDED
 #define QARCHIVE_MEMORY_EXTRACTOR_OUTPUT_HPP_INCLUDED
 #include <QObject>
-#include <QVector>
 
 #include <memory>
 
-#include "qarchivememoryfile.hpp"
 #include "qarchive_global.hpp"
+#include "qarchivememoryfile.hpp"
 
 namespace QArchive {
 class QARCHIVE_EXPORT MemoryExtractorOutput : public QObject {
     using QObject::QObject;
     Q_OBJECT
-  public:
-    MemoryExtractorOutput(std::unique_ptr<QVector<MemoryFile>>, QObject *parent = nullptr);
+public:
+    explicit MemoryExtractorOutput(std::unique_ptr<std::vector<MemoryFile>>, QObject* parent = nullptr);
 
-    [[gnu::warn_unused_result]] const QVector<MemoryFile> &getFiles() const;
+    [[gnu::warn_unused_result]] const std::vector<MemoryFile>& getFiles() const;
```

Comment:
This breaks the public API, we might want to keep that for the next release. Also is it bad using QVector? for now I think we will have a method like ```getFilesStdVector()``` for this implementation, we might remove the QVector version in the next release and rename this.

Reply:
Good point. std::vector allows using emplace_back. append() for some strange reason requires a defaulted constructor.